website/docs/components/copy/button/partials/guidelines/guidelines.md

```diff
@@ -35,3 +35,42 @@ Placing a Copy Button close to the content explicitly communicates what is being
 
 ![Example of the copy button component using proper placement](/assets/components/copy/copy-button-placement-do.png)
 !!!
+
+## Composition with other components
+
+Since the CopyButton can target almost any associated DOM element, it can be composed with other Helios components to create and support more complex scenarios and use cases.
+
+### With form inputs
+
+When combining the CopyButton with form inputs:
+
+- Use the [Fieldset](/components/form/primitives?tab=code#formfieldset) form primitive.
+- Wrap the `Base` input control (generally a TextInput, MaskedInput, or Textarea) and CopyButton in a container with an 8px gap between each element. This is not supported by the component out of the box, but can accomplished using a custom layout wrapper.
+- Use the `isIconOnly` variant of the CopyButton as there is a enough visual association using this layout method.
+
+![Example spacing of a control and CopyButton](/assets/components/copy/copy-button-composition-input-spacing.png)
+
+This ensures that other form primitives (helper text, error, etc) will span the combined width of the input control and the CopyButton.
+
+!!! Info
+
+Visit the [code](/components/copy/button?tab=code#composition-with-form-inputs) tab for contextual examples of this type of composition.
+!!!
+
```

Comment:
Is this really needed? It's the same for all the other components.

```suggestion
```

Reply:
Removed in https://github.com/hashicorp/design-system/pull/1876/commits/032c889aa8683a810599444fb7ecb63f452edc2c